### ISSUE TITLE
Warning thrown once maintenance file deleted

### ISSUE THREAD

**Issue by mattattui** [atst/stack-backstage#3]
With the PHP-FPM sapi (possibly with others), the PHP stat cache persists at least long enough that the realpath test in https://github.com/atst/stack-backstage/blob/master/src/Atst/StackBackstage.php#L21 will return true after the file is removed/renamed, resulting in the subsequent `file_get_contents()` call throwing an error.

I'm not sure what the best solution is. One is to `clearstatcache(true, $this->path)` just before `realpath` - but even though it's only clearing one file's realpath stats, it might still have a performance penalty.

Happy to submit a pull request if this solution is acceptable.

**Comment by mattattui** [comment]
Anyone there? :)

**Comment by davedevelopment** [comment]
Sorry Matt, I looked at the time and didn't have a solution. Calling `clearstatcache` would have to be a last resort for me, perhaps we should do it inside the conditional? Make's it messy though.

**Comment by mattattui** [comment]
Yeah it doesn’t strike me as particularly elegant either, though it’s not the whole stat cache, at least.

The only other way I can think of dealing with it is by telling users to delete the maintenance file using PHP (`unlink()` then `clearstatcache()`) rather than just removing the file from the command-line.

**Comment by davedevelopment** [comment]
That wouldn't clear the stat cache for all fpm workers though, only the one that happened to do the unlinking.

**Comment by mattattui** [comment]
Well… _poop_. Short of restarting php-fpm, I’m out of ideas!

**Comment by davedevelopment** [comment]
Let's go with the inelegant solution then!

``` php
$path = realpath($this->path);

if (false !== $path) {
    clearstatcache(true, $path);

    if (false !== realpath($path)) {
        return new Response(file_get_contents($path), 503);
    }
}
```

**Comment by mattattui** [comment]
Works for me! And no additional load when out of maintenance is plenty elegant enough for me.

Thanks <3
